aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/eventsourcing/source/core/SnapshotAwareEventSourceProducerTest.java

```diff
@@ -29,33 +31,54 @@
 @RunWith(MockitoJUnitRunner.class)
 public class SnapshotAwareEventSourceProducerTest {
 
+
     @Mock
     private QualifierAnnotationExtractor qualifierAnnotationExtractor;
 
     @Mock
     private EventSourceDefinitionRegistry eventSourceDefinitionRegistry;
 
-    @Mock
-    private JndiDataSourceNameProvider jndiDataSourceNameProvider;
-
     @Mock
     private SnapshotAwareEventSourceFactory snapshotAwareEventSourceFactory;
 
     @InjectMocks
     private SnapshotAwareEventSourceProducer snapshotAwareEventSourceProducer;
 
     @Test
-    public void shouldCreateEventSourceUsingTheJNDINameInjectedByTheContainer() throws Exception {
+    public void shouldCreateDefaultEventSourceDefinitionWhenEventSourceNameIsEmpty() throws Exception {
+        final EventSourceDefinition eventSourceDefinition = eventSourceDefinition()
+                .withName("defaultEventSource")
+                .withDefaultEventSource(true)
+                .withLocation(new Location("", "", Optional.of("dataSource")))
+                .build();
+        final InjectionPoint injectionPoint = mock(InjectionPoint.class);
+        final EventSourceName eventSourceNameAnnotation = mock(EventSourceName.class);
 
-        final String jndiName = "java:/app/my-command-api/DS.eventstore";
-        final String eventSourceName = "defaultEventSource";
+        final JdbcBasedEventSource jdbcBasedEventSource = mock(JdbcBasedEventSource.class);
+        when(qualifierAnnotationExtractor.getFrom(injectionPoint, EventSourceName.class)).thenReturn(eventSourceNameAnnotation);
+        when(eventSourceNameAnnotation.value()).thenReturn("");
+        when(eventSourceDefinitionRegistry.getDefaultEventSourceDefinition()).thenReturn(eventSourceDefinition);
+        when(snapshotAwareEventSourceFactory.create(eventSourceDefinition.getLocation().getDataSource().get(), eventSourceDefinition.getName())).thenReturn(jdbcBasedEventSource);
+
+        assertThat(snapshotAwareEventSourceProducer.eventSource(), is(jdbcBasedEventSource));
+    }
+
+    @Test
+    public void shouldCreateEventSourceUsingTheJNDINameInjectedByTheContainer() throws Exception {
```

Comment:
The test name does not seem to correspond with what is occurring in the test

Reply:
Test name changed